flytekit/core/type_engine.py

```diff
@@ -731,6 +734,30 @@ def to_python_value(self, ctx: FlyteContext, lv: Literal, expected_python_type:
         return expected_python_type(lv.scalar.primitive.string_value)
 
 
+def convert_json_schema_to_python_class(schema):
+    """Generate a model class based on the provided JSON Schema
+    :param schema: dict representing valid JSON schema
+    """
+    schema = copy.deepcopy(schema)
+
+    class Model(dict):
+        def __init__(self, *args, **kwargs):
+            self.__dict__["schema"] = schema
+            d = dict(*args, **kwargs)
+            dict.__init__(self, d)
+
+        def __setitem__(self, key, value):
```

Comment:
why should we support setitem?

Reply:
yeah, we don't use it now. I will remove it, and we can add it in the future if we need it.